resolwe_bio/processes/go_enrichment/go.yml

```diff
@@ -45,16 +44,26 @@
     - name: terms
       label: Enriched terms
       type: basic:json
+    - name: source
+      label: source
+      type: basic:string
   run:
     runtime: polyglot
     language: bash
     program: |
-      if [[ -n "{{ orthologues }}" ]]; then
-        orthologues.py {{ orthologues.ort.file }} {{ genes | join(" ") }} > temp.txt
+      SPECIES=`get_species.py "{{source}}" {{genes|join(' ')}}`
+      re-checkrc
+
+      if [ "${SPECIES}" == "{{gaf.species}}" ] && [ "{{source}}" == "{{gaf.source}}" ]; then
+        echo {{genes|join(" ")}} > genes.txt
       else
-        echo {{genes|join(" ")}} > temp.txt
+        map_features.py "{{source}}" "{{gaf.source}}" --feature_ids {{genes|join(' ')}}
+        if [ ! -f genes.txt ]; then
+          re-error "Input genes did not map to GAF gene IDs."
+        fi
       fi
 
-      processor {{pval_threshold}} {{min_genes}} {{ontology.obo_obj.file}} {{gaf.gaf_obj.file}} temp.txt 1>go_terms.txt 2> /dev/null
+      processor {{pval_threshold}} {{min_genes}} {{ontology.obo_obj.file}} {{gaf.gaf_obj.file}} genes.txt 1>go_terms.txt 2> /dev/null
```

Comment:
One problem we may have is that the command line will be too long for many genes. The same problem is in existing script in lines 58 and 60. Consider iterating over the list of genes and writing them to a file. Then specify the file in `goea.py` parameters.